python/pyspark/ml/torch/distributor.py

```diff
@@ -744,7 +814,99 @@ def run(self, train_object: Union[Callable, str], *args: Any) -> Optional[Any]:
                 TorchDistributor._run_training_on_pytorch_function  # type: ignore
             )
         if self.local_mode:
-            output = self._run_local_training(framework_wrapper_fn, train_object, *args)
+            output = self._run_local_training(framework_wrapper_fn, train_object, *args, **kwargs)
         else:
-            output = self._run_distributed_training(framework_wrapper_fn, train_object, *args)
+            output = self._run_distributed_training(
+                framework_wrapper_fn, train_object, None, *args, **kwargs
+            )
         return output
+
+    def train_on_dataframe(self, train_function, spark_dataframe, *args, **kwargs):
```

Comment:
It makes sense to create a new API since we are including `spark_dataframe` here, but the original premise was having a singular `.run` function be created to handle all forms of input. I guess I am not opposed to it, but can you add an example [here](https://github.com/apache/spark/blob/06e3b53ef432bdb4bab23c9204f4cbe096803f57/python/pyspark/ml/torch/distributor.py#L268) to showcase how it would look like?

Reply:
I guess something simple like [this](https://github.com/apache/spark/pull/40724/files#diff-364bfdd016cf2ded77548215dcba57f41060e5a5d2a7bcbdc309a8c5ccd4aa4bR78) would suffice.

Reply:
Sure ! I will add an example

Reply:
The main reason that I create a new API instead of adding new argument to `TorchDistributor.run` is:

`TorchDistributor.run` argument list ends with `*args` (args are arguments that passed to train_object), so that we have no way to add new argument without breaking change